### PR TITLE
[REG-1846] SDK Bug Fixes for Blue Flower Testing

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Fixes various SDK bugs / improvement areas found in blue flower testing
  - Handles uncaught bot segment errors better by ending the bot segment
  - Fixes a case where bot action errors weren't clearing their on screen error once they succeeded
  - Removes unused `isPressed` field from KeyboardInputActionData
    - Also re-orders the botsegment recording field order for easier editing
  - Renames `keyFrameCriteria` to `endCriteria` (backwards compatible) for bot segments due to repeated mis-understanding of when those criteria are evaluated... Hopefully the new name helps self explain that this is the end condition for the segment. 
  - Adds support for null/empty criteria bot segments

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

**Review by svolokh — APPROVED** [review]
Looks good to me!

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/JsonConverters/BotSegmentJsonConverter.cs [hunk 1]
```diff
@@ -26,7 +26,21 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             botSegment.sessionId = jObject.GetValue("sessionId")?.ToObject<string>(serializer) ?? Guid.NewGuid().ToString("n");
             botSegment.apiVersion = jObject.GetValue("apiVersion").ToObject<int>(serializer);
             botSegment.botAction = jObject.GetValue("botAction").ToObject<BotAction>(serializer);
-            botSegment.keyFrameCriteria = new List<KeyFrameCriteria>(jObject.GetValue("keyFrameCriteria").ToObject<KeyFrameCriteria[]>(serializer));
+            if (jObject.ContainsKey("keyFrameCriteria"))
+            {
+                // backwards compatibility before we renamed keyFrameCriteria to endCriteria
+                botSegment.endCriteria = new List<KeyFrameCriteria>(jObject.GetValue("keyFrameCriteria").ToObject<KeyFrameCriteria[]>(serializer));
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/JsonConverters/BotSegmentJsonConverter.cs:32]: Thanks for adding this backwards compatibility!

**Review by vontell — APPROVED** [review]
